File: SkoloInstituteAPI/SkoloInstitute/Entities/DataTransferObjects/Teacher/TeacherDto.cs
using SkoloInstitute.Entities.DataTransferObjects.Rating;
using SkoloInstitute.Entities.DataTransferObjects.Subject;

namespace SkoloInstitute.Entities.DataTransferObjects.Teacher
{
    public class TeacherDto
    {
        public Guid Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public string PictureUrl { get; set; }
        public int Rating { get; set; }
        public IEnumerable<SubjectDto> Subjects { get; set; }
        public IEnumerable<RatingDto> Ratings { get; set; }

    }
}

File: skolo-ui/src/app/_interfaces/teacher/TeacherDto.ts
import { RatingDto } from "../rating/RatingDto";
import { SubjectDto } from "../subject/SubjectDto";

export interface TeacherDto {
    id: string; // Use string for GUIDs in TypeScript
    firstName: string;
    lastName: string;
    phone: string;
    email: string;
    pictureUrl: string;
    rating: number;
    subjects: SubjectDto[];
    ratings?:RatingDto;
  }